src/osd/PrimaryLogPG.cc

```diff
@@ -15617,8 +15617,10 @@ PrimaryLogPG::AwaitAsyncWork::AwaitAsyncWork(my_context ctx)
     NamedState(nullptr, "Trimming/AwaitAsyncWork")
 {
   auto *pg = context< SnapTrimmer >().pg;
+  // Determine cost in terms of the average object size
+  uint64_t cost_per_object = pg->get_average_object_size();
   context< SnapTrimmer >().log_enter(state_name);
-  context< SnapTrimmer >().pg->osd->queue_for_snap_trim(pg);
+  context< SnapTrimmer >().pg->osd->queue_for_snap_trim(pg, cost_per_object);
```

Comment:
Does each snap trim item trim a specific number of objects?  This should probably be multiplied by the number of objects each queue iteration actually trims.  Otherwise, this branch looks great -- thanks for the commit factoring out the helper!

Reply:
@athanatos The number of objects to trim in an iteration is bound by osd_pg_max_concurrent_snap_trims which is currently set to 2 by default.  However, currently the actual number of objects to trim is determined using SnapMapper after the item is dequeued as part of `AwaitAsyncWork::react(const DoSnapWork&)`. Therefore it looks like the number objects to trim could range between 1 and 2 for each iteration.

Possible Solution:
For the cost to reflect the actual number of objects that would be trimmed, the  objects to trim must be predetermined, or in other words before `DoSnapWork`. For this,  `AwaitAsyncWork` state would need to maintain the vector of objects to trim (`to_trim`). Once all the objects for the iteration are trimmed, I think a new OpSchedulerItem called `PGSnapTrimFinish` with min cost can be enqueued by `AwaitAsyncWork` which would invoke a new method called `FinishSnapWork` to perform the clean-up and move the state to `NotTrimming`.

There are other details I may have missed, but do you think the above approach seems reasonable?

Reply:
I think this is the best approach -- we don't need to make it more complicated than this.